projects/packages/videopress/src/client/admin/components/video-row/index.tsx

```diff
@@ -0,0 +1,263 @@
+			onFocus={ () => setShowPopover( true ) }
+			onBlur={ () => setShowPopover( false ) }
+			className={ className }
+		>
+			<Icon icon={ icon } />
+			{ showPopover && <PopoverWithAnchor anchorRef={ anchorRef }>{ children }</PopoverWithAnchor> }
+		</div>
+	);
+};
+
+const QuickActions = ( { button }: { button: React.ReactNode } ) => {
+	// Hiding it based on Design request:
+	// https://github.com/Automattic/jetpack/issues/25742#issuecomment-1223123815
+	const HIDE_QUICK_ACTIONS = false;
+
+	if ( HIDE_QUICK_ACTIONS ) {
+		return null;
+	}
+
+	return (
+		<div className={ styles.actions }>
+			{ button }
+			<ActionItem icon={ image }>{ __( 'Update thumbnail', 'jetpack-videopress-pkg' ) }</ActionItem>
+			<ActionItem icon={ privacy }>{ __( 'Update privacy', 'jetpack-videopress-pkg' ) }</ActionItem>
+			<ActionItem icon={ trash } className={ styles.trash }>
+				{ __( 'Delete video', 'jetpack-videopress-pkg' ) }
+			</ActionItem>
+		</div>
+	);
+};
+
+const Stats = ( {
+	duration,
+	uploadDate,
+	plays,
+	isPrivate,
+}: {
+	duration: string;
+	uploadDate: string;
+	plays: number;
+	isPrivate: boolean;
+} ) => {
+	const [ isSmall ] = useBreakpointMatch( 'sm' );
+	const durationLabel = __( 'Duration', 'jetpack-videopress-pkg' );
+	const playsLabel = __( 'Plays', 'jetpack-videopress-pkg' );
+	const privacyLabel = __( 'Privacy', 'jetpack-videopress-pkg' );
+
+	return (
+		<div className={ classNames( styles.stats, { [ styles.small ]: isSmall } ) }>
+			<div className={ styles.privacy } aria-disabled={ isSmall ? 'false' : 'true' }>
+				{ isPrivate && ! isSmall && <Icon icon={ privacy } /> }
+				{ isSmall && (
+					<>
+						<span>{ privacyLabel }</span>
+						<span>
+							{ isPrivate
+								? __( 'Private', 'jetpack-videopress-pkg' )
+								: __( 'Public', 'jetpack-videopress-pkg' ) }
+						</span>
+					</>
+				) }
+			</div>
+			<div className={ styles.duration }>
+				<span>{ isSmall && durationLabel }</span>
+				<span>{ duration }</span>
+			</div>
+			{ Number.isFinite( plays ) && (
+				<div className={ styles.plays }>
+					<span>{ isSmall && playsLabel }</span>
+					<span>{ plays }</span>
+				</div>
+			) }
+			{ ! isSmall && <div className={ styles.upload }>{ uploadDate }</div> }
+		</div>
+	);
+};
+
+const VideoRow = ( {
+	videoTitle,
+	posterImage,
+	duration,
+	uploadDate,
+	plays = null,
+	isPrivate = false,
+	onClickEdit,
+}: {
+	videoTitle: string;
+	posterImage: string;
+	duration: number;
+	uploadDate: string;
+	plays: number;
+	isPrivate: boolean;
+	onClickEdit?: () => void;
+} ) => {
+	const [ isSmall ] = useBreakpointMatch( 'sm' );
+	const [ showActions, setShowActions ] = useState( false );
+	const [ keyPressed, setKeyDown ] = useState( false );
+	const [ expanded, setExpanded ] = useState( false );
+	const textRef = useRef( null );
+
+	const durationInMinutesAndSeconds = millisecondsToMinutesAndSeconds( duration );
+	const uploadDateFormatted = dateI18n( 'M j, Y', uploadDate, null );
```

Comment:
Question: Is this format `M j, Y` applied in all languages?

Reply:
> @retrofox about the design part, there’s inconsistencies across figma, I’ll reach out design and we can improve (if needed) later. Let’s follow this way for now.

Where there are the inconsistencies? The row height is 90px in all instances that I took a look at, for instance.
Why we are going to implement something that is not consistent with the design? Even more, we can clarify the inconsistencies just here, asking designer folks. This is the proper moment to do it.

Reply:
Good question. Don't know exactly, but we could figure out.

Reply:
Good catch. What we usually do is use the date format defined in the site settings

Reply:
@leogermani do you know how we do at client code?